sycl/doc/extensions/proposed/sycl_ext_oneapi_complex_marray.asciidoc

```diff
@@ -0,0 +1,543 @@
+
+  template <class T, std::size_t NumElements, std::size_t... I>
+  marray<sycl::ext::oneapi::complex<T>, NumElements> make_complex_marray(const marray<sycl::ext::oneapi::complex<T>, NumElements> &cmplx, std::integer_sequence<std::size_t, I...> int_seq);
+
+  // Get
+
+  // return marray of component
+  template <class T, std::size_t NumElements>
+  marray<T, NumElements> get_real(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input);
+
+  template <class T, std::size_t NumElements>
+  marray<T, NumElements> get_imag(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input);
+
+  // return element of component
+  template <class T, std::size_t NumElements>
+  T get_real(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input, std::size_t index);
+
+  template <class T, std::size_t NumElements>
+  T get_imag(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input, std::size_t index);
+
+  // return sequence of elements of component
+  template <class T, std::size_t NumElements, std::size_t... I>
+  T get_real(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input, std::integer_sequence<std::size_t, I...> int_seq);
+
+  template <class T, std::size_t NumElements, std::size_t... I>
+  T get_imag(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input, std::integer_sequence<std::size_t, I...> int_seq);
+
+  // Set
+
+  template <class T, std::size_t NumElements>
+  void set_real(marray<sycl::ext::oneapi::complex<T>, NumElements> &input, const marray<T, NumElements> &values);
+
+  template <class T, std::size_t NumElements>
+  void set_imag(marray<sycl::ext::oneapi::complex<T>, NumElements> &input, const marray<T, NumElements> &values);
+
+  template <class T, std::size_t NumElements>
+  void set_real(marray<sycl::ext::oneapi::complex<T>, NumElements> &input, const T value);
+
+  template <class T, std::size_t NumElements>
+  void set_imag(marray<sycl::ext::oneapi::complex<T>, NumElements> &input, const T value);
+
+  template <class T, std::size_t NumElements>
+  void set_real(marray<sycl::ext::oneapi::complex<T>, NumElements> &input, std::size_t index, const T value);
+
+  template <class T, std::size_t NumElements>
+  void set_imag(marray<sycl::ext::oneapi::complex<T>, NumElements> &input, std::size_t index, const T value);
+
+} // namespace oneapi
+} // namespace ext
+} // namespace sycl
+```
+
+The class `sycl::oneapi::marray<complex<T>, N>`, has specializations
+of `T`; `float`, `double`, and `sycl::half` defined.
+
+```C++
+namespace sycl {
+namespace ext {
+namespace oneapi {
+
+  template <std::size_t NumElements>
+  class marray<double, NumElements>;
+
+  template <std::size_t NumElements>
+  class marray<float, NumElements>;
+
+  template <std::size_t NumElements>
+  class marray<sycl::half, NumElements>;
```

Comment:
What are these for? Are they missing `complex`?

Reply:
This was a typo, these should be specialized with `complex<sycl::half>`, etc. I have updated it.